core/geometry/src/curve/internalContexts/geometry2d/UnboundedCircle2d.ts

```diff
@@ -0,0 +1,185 @@
+  public center: Point2d;
+  /** The circle radius */
+  public radius: number;
+  /* Constructor - CAPTURE given center and normal */
+  private constructor(center: Point2d, radius: number) {
+    super();
+    this.center = center;
+    this.radius = radius;
+  }
+    /**
+     * Return a clone of this circle.
+     */
+    public clone () : UnboundedCircle2dByCenterAndRadius {
+      // (The create method clones the inputs . . .)
+      return  UnboundedCircle2dByCenterAndRadius.createPointRadius (this.center, this.radius);
+    }
+    /**
+     * Return a clone of this circle, with radius negated
+     */
+    public cloneNegateRadius () : UnboundedCircle2dByCenterAndRadius {
+      // (The create method clones the inputs . . .)
+      return  UnboundedCircle2dByCenterAndRadius.createPointRadius (this.center, -this.radius);
+    }
+  /**
+   * Create an ImplicitCircle2d from XY parts of its center and its radius
+   * @param centerX x coordinate of center
+   * @param centerY y coordinate of center
+   * @param radius circle radius
+   * @returns newly created circle object
+   */
+  public static createXYRadius(centerX: number, centerY: number, radius: number): UnboundedCircle2dByCenterAndRadius {
+    return new UnboundedCircle2dByCenterAndRadius(Point2d.create(centerX, centerY), radius);
+  }
+
+  /**
+   * Create an ImplicitCircle2d from an xy object and a radius.
+   * * Zero radius is valid.
+   * * The input coordinates are copied -- the center is NOT captured.
+   * @param center xy coordinates of center
+   * @param radius circle radius
+   * @returns newly created circle object
+   */
+  public static createPointRadius(center: XAndY, radius: number): UnboundedCircle2dByCenterAndRadius {
+    return new UnboundedCircle2dByCenterAndRadius(Point2d.create(center.x, center.y), radius);
+  }
+  /**
+   *
+   * @param xy space paoint
+   * @returns squared distance to center minus squared radius
+   */
+public override functionValue (xy: XAndY) : number {
+  return Geometry.distanceSquaredXYXY (xy.x, xy.y, this.center.x, this.center.y) - this.radius * this.radius;
+}
+  /**
+   *
+   * @param xy space paoint
+   * @returns gradient of the implicit function.
+   */
+  public override gradiant (xy: XAndY) : Vector2d {
+    return Vector2d.create(2 * (xy.x - this.center.x), 2 * (xy.y - this.center.y));
+  }
+
+  /**
+   * Emit circle points for which a vector to the space point is perpendicular to the circle.
+   * * For a non-zero radius circle, there are two perpendiculars.  The one on the side of the space point is emitted first.
+   * * For a zero radius circle, the vector from center to the space point is the only perpendicular
+   * @param spacePoint
+   * @param handler
+   */
+public override emitPerpendiculars(spacePoint: Point2d,
+   handler :(curvePoint: Point2d, radians: number | undefined)=>any):any{
+  const radialVector = Vector2d.createStartEnd (this.center, spacePoint).scaleToLength (this.radius);
```

Comment:
`radialVector` is just the radius line from circle center to circle circumference toward the spacePoint. How the 2 calls to `handler` on line 95-96 finds the perp line from spacePoint to the circle?